subprojects/docs/src/docs/userguide/javaTesting.adoc

```diff
@@ -398,3 +443,42 @@ Creating and configuring a source set automatically sets up the compilation stag
 ++++
 
 Again, we're accessing a source set to get the relevant information, i.e. where the compiled test classes are — the `testClassesDir` property — and what needs to be on the classpath when running them — `classpath`.
+
+Users commonly want to run integration tests after the unit tests, because they are often slower to run and you want the build to fail early on the unit tests rather than later on the integration tests. That's why the above example adds a `shouldRunAfter()` declaration. This is preferred over `mustRunAfter()` so that Gradle has more flexibility in executing the build in parallel.
+
+[[sec:skipping_java_tests]]
+=== Skipping the tests
+
+If you want to skip the tests when running a build, you have a few options. You can either do it via <<sec:excluding_tasks_from_the_command_line,command line arguments>> or <<sec:skipping_tasks,in the build script>>. To do it on the command line, you can use the `-x` or `--exclude-task` option like so:
+
+    gradle build -x test
+
+This excludes the `test` task and any other task that it _exclusively_ depends on, i.e. no other task depends on the same task. Those tasks will not be marked "SKIPPED" by Gradle, but will simply not appear in the list of tasks executed.
+
+Skipping a test via the build script can be done a few ways. One common approach is to make test execution conditional via the `onlyIf()` method. The following sample skips the `test` task if the project has a property called `mySkipTests`:
+
+++++
+<sample id="skipTestsInBuildScript" dir="userguide/java/basic" title="Skipping the unit tests based on a project property">
+    <sourcefile file="build.gradle" snippet="skip-tests-condition"/>
+</sample>
+++++
+
+In this case, Gradle will mark the skipped tests as "SKIPPED" rather than exclude them from the build.
+
+[[sec:forcing_java_tests_to_run]]
+=== Forcing tests to run
+
+In well-defined builds, you can rely on Gradle to only run tests if the tests themselves or the production code change. However, you may encounter situations where the tests rely on a third-party service or something else that might change but can't be modeled in the build.
+
+You can force tests to run in this situation by cleaning the output of the relevant `Test` task — say `test` — and running the tests again, like so:
+
+    gradle cleanTest test
+
+`cleanTest` is based on a _<<sec:task_rules,task rule>>_ provided by the <<sec:base_tasks,Base Plugin>>. You can use it for _any_ task.
```

Comment:
❓ is there a place a user can learn about how to use this useful task rule?

Reply:
Beyond the Base Plugin chapter? Not that I'm aware of. Unfortunately, there isn't an anchor specifically for `clean<Task>` in that chapter either, but it's not hard to find.